src/main/java/com/mg105/usecase/Inventory/InventoryInteractor.java

```diff
@@ -0,0 +1,156 @@
+package com.mg105.usecase.Inventory;
+
+import com.mg105.entities.BattleCharacter;
+import com.mg105.entities.GameState;
+import com.mg105.entities.Inventory;
+import com.mg105.outputds.ItemDetails;
+import com.mg105.presenterinterfaces.InventoryPresenterInterface;
+import com.mg105.utils.ItemConstants;
+import com.mg105.utils.PartyConstants;
+
+import java.util.NoSuchElementException;
+
+public class InventoryInteractor {
+
+    /**
+     * This class is the only class that should directly interact with the inventory
+     * <p>
+     * It is responsible for all things related to get information from the inventory and updating the
+     * inventory.
+     */
+
+    private final GameState gameState;
+
+    private final ItemFactory itemFactory = new ItemFactory();
+
+    private final InventoryPresenterInterface response;
+
+    public InventoryInteractor(GameState gamestate, InventoryPresenterInterface response) {
+        this.gameState = gamestate;
+        this.response = response;
+    }
+
+    /**
+     * Returns an object that represents if an item was successfully added
+     *
+     * @param itemName the name of the item to be added
+     * @return the state of the item in the inventory and a description on if the item was added
+     * @see InventoryPresenterInterface
+     */
+
+    public String[] addItem(String itemName) {
```

Comment:
Yeah, I agree that this should prob just be a boolean representing success or failure